src/share.tsx

```diff
@@ -17,10 +19,65 @@ function Header(props: PropsWithChildren): JSX.Element {
   )
 }
 
-export function ShareSpace (): JSX.Element {
+function isDID(value: string): boolean {
+  try {
+    DID.parse(value.trim())
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+function isEmail(value: string): boolean {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  return !isDID(value) && emailRegex.test(value)
```

Comment:
Could you use the didmailto module's `email` method here?

Reply:
Do you know which method of that module? The one I found `DIDMailTo.email` says it was not supposed to be used as an email validator. Is it correct?

Reply:
Oh, ok no worries then...